AdminCore.WebApp/Controllers/ProjectController.cs

```diff
@@ -0,0 +1,95 @@
+﻿using AdminCore.Common.Interfaces;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using AdminCore.DTOs.Project;
+using AdminCore.WebApi.Models.Client;
+using AdminCore.WebApi.Models.Project;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+
+namespace AdminCore.WebApi.Controllers
+{
+  [ApiController]
+  [Route("[controller]")]
+  public class ProjectController : BaseController
+  {
+    private readonly IProjectService _projectService;
+
+    public ProjectController(IProjectService projectService, IMapper mapper) : base(mapper)
+    {
+      _projectService = projectService;
+    }
+
+    [HttpPost]
+    [ProducesResponseType(typeof(CreateProjectViewModel), StatusCodes.Status201Created)]
+    public IActionResult CreateProject([FromBody] CreateProjectViewModel projectToCreate)
+    {
+      try
+      {
+        var createdProject = _projectService.CreateProject(Mapper.Map<ProjectDto>(projectToCreate));
+        return Created(createdProject.ProjectId.ToString(), Mapper.Map<ProjectViewModel>(createdProject));
+      }
+      catch (DbUpdateException e)
+      {
+        return BadRequest();
+      }
+    }
+
+    [HttpPut]
+    [ProducesResponseType(typeof(IList<ProjectViewModel>), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public IActionResult UpdateProject([FromBody] UpdateProjectViewModel projectToUpdate)
+    {
+      try
```

Comment:
private IActionResult HandleBarRequest(Action updateAction) {
  try {
   action();
  }
  catch (DbUpdateException e)
  {
        return BadRequest();
  }
}

Reply:
Now using this Func<>